Datetime optimization

tsteven4 opened the pull request:
Prefer Qt::UTC TimeSpec for storing creation_time.
This avoids very significant performance hits from converting
to/from Qt::LocalTime, which involve DST calculations as well
as the usual offsets.
Test suite with this commit runs at 63% of wall-clock time of the
baseline (time ./testo)!    lowranceusr test runs at 37% of wall-clock time compared to the baseline.

Delete obsolete operators and methods of DateTime.

tsteven4 committed:
Optimize performance of DateTime.

Prefer Qt::UTC TimeSpec for storing creation_time.
This avoids very significant performance hits from converting
to/from Qt::LocalTime, which involve DST calculations as well
as the usual offsets.
Test suite with this commit runs at 63% of wall-clock time of the
baseline (time ./testo)!

tsteven4 committed:
remove obsolete fromTime_t and use Qt::UTC.

tsteven4 committed:
Delete obsolete operators and methods of DateTime.

robertlipe reviewed (approved):
I like this a lot. Go for it, please.

tsteven4 committed:
Merge branch 'master' into datetimeoptimization

tsteven4 committed:
document motivation for Qt::UTC as default timespec.

Modify Waypoint::SetCreationTime so it can be used with seconds, milliseconds,
or both operands.  This method is implemented without resetting the
Qt::TimeSpec.  It will be efficient if the existing TimeSpec is
Qt::UTC as set up by gpsbabel::DateTime default constructor.
